### DISCUSSION TITLE
Typing < on Linux Mint with Italian Keyboard

### DISCUSSION THREAD

**Discussion by pamoroso** [Interlisp/medley#1828] [General]
This is a followup to the discussion in the September 11, 2024 external meeting about my inability to type `<`, both in Medley Online and VNC, on my [Linux Mint PC](https://system76.com/desktops/meerkat) with an [Italian keyboard](https://i.snap.as/wAoI3ouY.jpg). When I press the `<` key nothing happens in Medley.

According to `xev` these are the events pressing the `<` key generates:

```
KeyPress event, serial 38, synthetic NO, window 0x3200001,
    root 0x5cf, subw 0x0, time 1419311, (91,50), root:(156,853),
    state 0x0, keycode 94 (keysym 0x3c, less), same_screen YES,
    XLookupString gives 1 bytes: (3c) "<"
    XmbLookupString gives 1 bytes: (3c) "<"
    XFilterEvent returns: False

KeyRelease event, serial 38, synthetic NO, window 0x3200001,
    root 0x5cf, subw 0x0, time 1419375, (91,50), root:(156,853),
    state 0x0, keycode 94 (keysym 0x3c, less), same_screen YES,
    XLookupString gives 1 bytes: (3c) "<"
    XFilterEvent returns: False
```

**Comment by nbriggs** [comment]
I would guess that there's no mapping because in the default read tables `<` is expected to be `shift+,` not a separate physical key.

**Reply by fghalasz** [reply]
Hmmm.  In that case, it looks like the + key on Paolo's keyboard should have the same problem.  On US keyboard + is a shifted key.  Looks like on Paolo's keyboard, + is a non-shifted key. 

Paolo?

**Comment by pamoroso** [comment]
Typing `+` by pressing the corresponding unshifted key works as expected on my Italian keyboard, both in VNC Medley and Medley Online.  And I can type all the other characters, shifted and unshifted (except of course for accented letters).

**Comment by pamoroso** [comment]
The [5th round of TEdit changes](https://files.interlisp.org/medley/library/tedit/TEDIT-RELEASENOTES.TEDIT.pdf) added support for `Meta` key combinations to move the caret backward (`<` or `,`) or forward (`>` or `.`) one character.

As expected, pressing `Meta-<` in TEdit on my Linux PC with [Italian keyboard](https://i.snap.as/wAoI3ouY.jpg) does nothing as reported in this issue. But, interestingly, pressing `Meta-,` doesn't do anything either. With the same keyboard typing `,` in TEdit works fine and doesn't even require `Shift`. `Meta->` and `Meta-.` work too.

It's a weird coincidence that Maiko and/or Medley seem blind to keychords that move the caret left on non English keyboards.

**Reply by pamoroso** [reply]
This is the information `xev` shows about the `,` keypress on my Italian keyboard:

```
KeyPress event, serial 28, synthetic NO, window 0x4200001,
    root 0x5cf, subw 0x4200002, time 39556239, (45,22), root:(110,825),
    state 0x0, keycode 59 (keysym 0x2c, comma), same_screen YES,
    XLookupString gives 1 bytes: (2c) ","
    XmbLookupString gives 1 bytes: (2c) ","
    XFilterEvent returns: False

KeyRelease event, serial 28, synthetic NO, window 0x4200001,
    root 0x5cf, subw 0x4200002, time 39556351, (45,22), root:(110,825),
    state 0x0, keycode 59 (keysym 0x2c, comma), same_screen YES,
    XLookupString gives 1 bytes: (2c) ","
    XFilterEvent returns: False
```

**Comment by pamoroso** [comment]
A good workaround for me is to remap keypad `0` to `<` unshifted and `0` shifted in my init file with `(KEYACTION 'KEYPAD0 '((60 49 NOLOCKSHIFT) . IGNORE))`.
